packages/plugin-npm/sources/npmHttpUtils.ts

```diff
@@ -0,0 +1,28 @@
+import {Configuration, httpUtils, Ident} from '@berry/core';
+
+export function get(url: string, ident: Ident, configuration: Configuration) {
+  const headers: {[headerName: string]: string} = {};
+
+  const auth = getAuthenticationHeader(ident, configuration);
+  if (auth)
+    headers.authorization = auth;
+
+  return httpUtils.get(
+      url,
+      configuration,
+      {headers},
+  );
+}
+
+function getAuthenticationHeader(ident: Ident, configuration: Configuration) {
+  if (!configuration.get(`npmAlwaysAuth`) && !ident.scope)
+    return undefined;
+
+  if (configuration.get(`npmAuthToken`))
+    return `Bearer ${configuration.get(`npmAuthToken`)}`;
+
+  if (configuration.get(`npmAuthIdent`))
+    return `Basic ${configuration.get(`npmAuthIdent`)}`;
+
+  throw new Error(`No authentication configured for request`);
+}
```

Comment:
@arcanis would it be fine for me to add an editorconfig to the repo? That file basically replaces the settings currently in `.vscode/settings.json`, but it also exposes options like "insert trailing newline", "strip trailing whitespace" etc.

For reference, the `.editorconfig` I use:

```
# EditorConfig is awesome!

# Mark this as the root editorconfig file
root = true

# Base ruleset for all files
[*]
trim_trailing_whitespace = true
insert_final_newline = true
indent_style = space
indent_size = 2
continuation_indent_size = 4

# Override rules for markdown
[*.md]
# trailing whitespace is significant in markdown -> do not remove
trim_trailing_whitespace = false
```

Reply:
Yep, sounds good to me 👍